libs/SalesforceSDKCore/SalesforceSDKCore/Classes/SmartStore/SFSmartStore.m

```diff
@@ -1607,12 +1607,8 @@ - (void) projectIndexedPaths:(NSDictionary*)entry values:(NSMutableDictionary*)v
     for (SFSoupIndex *idx in indices) {
         NSString *indexColVal = [SFJsonUtils projectIntoJson:entry path:[idx path]];
         NSString *colName = [idx columnName];
-        if (nil != indexColVal //not every entry will have a value for each index column
-            && (typeFilter == nil || [typeFilter isEqualToString:idx.indexType]))
-        {
-            values[colName] = indexColVal;
-        } else {
-            values[colName] = [NSNull null];
+        if (typeFilter == nil || [typeFilter isEqualToString:idx.indexType]) {
```

Comment:
Type filter is specified when preparing values to insert for full text table. We don't want to insert anything since the columns won't be there in the full text table.